gfx_scene.h

```diff
@@ -1110,473 +1166,267 @@ class GfxSceneInternal
-        std::map<int32_t, std::pair<GfxConstRef<GfxImage>, GfxConstRef<GfxImage>>> maps;
-        for(size_t i = 0; i < gltf_model.materials.size(); ++i)
-        {
-            std::map<int32_t, GfxConstRef<GfxImage>>::const_iterator it;
-            tinygltf::Material const &gltf_material = gltf_model.materials[i];
+            std::map<cgltf_image const*, GfxConstRef<GfxImage>>::const_iterator it;
+            cgltf_material const &gltf_material = gltf_model->materials[i];
+            if(!gltf_material.has_pbr_metallic_roughness)
+                continue; // unsupported material type
             GfxRef<GfxMaterial> material_ref = gfxSceneCreateMaterial(scene);
             GfxMetadata &material_metadata = material_metadata_[material_ref];
             material_metadata.asset_file = asset_file;  // set up metadata
             material_metadata.object_name = gltf_material.name;
             GfxMaterial &material = *material_ref;
-            parameter = GetParameter(gltf_material, "baseColorFactor");
-            if(parameter.first) material.albedo = glm::vec4(parameter.second);
-                           else material.albedo = glm::vec4(1.0f, 1.0f, 1.0f, 1.0f);
-            parameter = GetParameter(gltf_material, "roughnessFactor");
-            if(parameter.first) material.roughness = parameter.second.x;
-                           else material.roughness = 1.0f;
-            parameter = GetParameter(gltf_material, "metallicFactor");
-            if(parameter.first) material.metallicity = parameter.second.x;
-                           else material.metallicity = 1.0f;
-            parameter = GetParameter(gltf_material, "emissiveFactor");
-            if(parameter.first) material.emissivity = glm::vec3(parameter.second);
-                           else material.emissivity = glm::vec3(0.0f, 0.0f, 0.0f);
-            int32_t const albedo_map = GetTextureIndex(gltf_material, "baseColorTexture");
-            it = (albedo_map >= 0 ? images.find(albedo_map) : images.end());
+            cgltf_pbr_metallic_roughness const &gltf_material_pbr = gltf_material.pbr_metallic_roughness;
+            material.albedo = glm::vec4(gltf_material_pbr.base_color_factor[0], gltf_material_pbr.base_color_factor[1],
+                gltf_material_pbr.base_color_factor[2], gltf_material_pbr.base_color_factor[3]);
+            material.roughness = gltf_material_pbr.roughness_factor;
+            material.metallicity = gltf_material_pbr.metallic_factor;
+            material.emissivity = glm::vec3(gltf_material.emissive_factor[0], gltf_material.emissive_factor[1],
+                gltf_material.emissive_factor[2]);
+            cgltf_texture const *albedo_map_text = gltf_material_pbr.base_color_texture.texture;
+            it = (albedo_map_text != nullptr ? images.find(albedo_map_text->basisu_image != nullptr ?
+                albedo_map_text->basisu_image : albedo_map_text->image) : images.end());
```

Comment:
Since compressed textures are an option extension it is possible for a gltf file to have both a compressed image (```basisu_image```) and a normal image (```image```). So since we support compressed we use that first if it is specified otherwise load the regular image.

Reply:
Sounds good.